agent/app/agent.go

```diff
@@ -176,8 +188,30 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 		return exitcodes.ExitTerminal
 	}
 
+	var vpcSubnetAttributes []*ecs.Attribute
+	// Check if Task ENI is enabled
+	if agent.cfg.TaskENIEnabled {
+		if err, terminal := agent.initializeTaskENIDependencies(state, taskEngine); err != nil {
+			if err != instanceNotLaunchedInVPCError {
+				seelog.Criticalf("Unable to initialize Task ENI dependencies: %v", err)
+				if terminal {
+					return exitcodes.ExitTerminal
+				}
+				return exitcodes.ExitError
+			}
+			// We have ascertained that the EC2 Instance is not running in a VPC
+			// No need to stop the ECS Agent in this case; all we need to do is
+			// to not update the config to disabled the TaskENIEnabled flag and
+			// move on
+			seelog.Warnf("Unable to detect VPC ID for the Instance, disabling Task ENI capability: %v", err)
+			agent.cfg.TaskENIEnabled = false
```

Comment:
I think the use-case here is that a customer running an instance with ECS Agent installed via the `ECS Init` package in non-VPC land would have their instance never register because it's enabled via ECS Init. The right thing to do here may be to disable this using the config variable via ECS Init. But then, you'd have to deal with failure cases where customers might have created the `ecs.config` file with this turned on. So, disabling this on instances not running on VPC seems like the right answer. Let me know if you have alternate suggestions here.